bigquery_etl/schema/__init__.py

```diff
@@ -247,7 +247,7 @@ def _traverse(
                                     f"for {prefix}.{field_path} are incompatible"
                                 )
 
-                if dtype == "RECORD":
+                if dtype == "RECORD" and nodes[node_name]["type"] == "RECORD":
```

Comment:
Did an issue with this pop up while you were testing?  I'm `ignore_incompatible_fields` would need to be true

Reply:
> Did an issue with this pop up while you were testing?

Yes, it was causing the `derived_view_schemas` generator to fail for a couple views that transform an underlying struct column into something else:
- `telemetry.clients_daily_joined`
- `telemetry.clients_daily_v6`

> I'm `ignore_incompatible_fields` would need to be true

The `derived_view_schemas` generator calls `Schema.merge()` with `attributes=["description"]`, which kind of a has a similar effect since the field incompatibility logic is applied at the attribute level.